packages/admin/admin-icons/generate-icons.ts

```diff
@@ -6,48 +6,65 @@ import { pascalCase, pascalCaseTransformMerge } from "pascal-case";
 import * as path from "path";
 const eslint = new ESLint({ fix: true });
 
+type Icon = {
+    name: string;
+    path: string;
+    componentName: string;
+    deprecated: boolean;
+};
+
 const main = async () => {
-    const files = readdirSync("icons").filter((file) => {
-        return path.extname(file).toLowerCase() === ".svg";
-    });
+    const isSvg = (file: string) => path.extname(file).toLowerCase() === ".svg";
+
+    const iconFiles = readdirSync("icons").filter(isSvg);
+    const deprecatedIconFiles = readdirSync("icons/deprecated").filter(isSvg);
+
+    const icons = [
```

Comment:
use type annotation here:
```suggestion
    const icons: Icon[] = [
```

Reply:
dd41ab5dd0eceefbaf23fec2b044cf9995bd1e68